lib/filterx/expr-compound.c

```diff
@@ -84,6 +84,11 @@ _eval_exprs(FilterXCompoundExpr *self, FilterXObject **result)
   for (gint i = 0; i < self->exprs->len; i++)
     {
       filterx_object_unref(*result);
+      FilterXEvalContext *context = filterx_eval_get_context();
+
+      if (context->eval_control_type != FXC_NOTSET)
+        /* code flow modifier detected, short circuiting */
```

Comment:
[nitpick]

Maybe we should say "control flow modifier", to match the variable name.